src-self-hosted/link.zig

```diff
@@ -38,8 +33,12 @@ pub const Options = struct {
     /// Used for calculating how much space to reserve for executable program code in case
     /// the binary file deos not already have such a section.
     program_code_size_hint: u64 = 256 * 1024,
+    default_entry_addr: u64 = 0x8000000,
```

Comment:
Hmm, if you're moving non-ELF constants and structures around, could you please update the `MachO.zig` as well?

Reply:
The "default" here is `0x8000000`. If you want to add a field to these options it would be `entry_addr`. A default is what happens when no explicit choice is made.

```suggestion
    entry_addr: ?u64 = null,
```

And leave the global alone. Then later you can make your choice, `options.entry_addr orelse if (spu2) 0 else default_entry_addr`